packages/aws-cdk-lib/aws-ec2/README.md

```diff
@@ -1547,6 +1547,36 @@ const aspect = new ec2.InstanceRequireImdsv2Aspect();
 Aspects.of(this).add(aspect);
 ```
 
+### Associating a Public IP Address with an Instance
+
+All subnets have an attribute that determines whether instances launched into that subnet are assigned a public IPv4 address. This attribute is set to true by default for default public subnets. Thus, an EC2 instance launched into a default public subnet will be assigned a public IPv4 address. Nondefault public subnets have this attribute set to false by default and any EC2 instance launched into a nondefault public subnet will not be assigned a public IPv4 address automatically. To automatically assign a public IPv4 address to an instance launched into a nondefault public subnet, you can set the `associatePublicIpAddress` property on the `Instance` construct to true. Alternatively, to not automatically assign a public IPv4 address to an instance launched into a default public subnet, you can set `associatePublicIpAddress` to false. Importantly, if you set `associatePublicIpAddress` and there is a VPC defined in the same stack, you must make sure the `Instance` has a dependency on the VPC to declare a dependency on the VPC-gateway attachment.
```

Comment:
Can you add a note here about what happens if you want to switch this value for an existing instance?

Reply:
I've updated this section in the README to include a note about what happens when associatePublicIpAddress is included, removed, or updated on an existing instance.